### ISSUE TITLE
User can select invisible element in modal component

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1474]
## Environment
Chrome 70
macOS Mojave

## Violation
If the user fires the modal they can tab through the elements logically -- but at one point they can also focus an invisible element. 

Calling `document.activeElement` while that element has focus returns the entire modal 
<img width="812" alt="screen shot 2018-11-25 at 6 07 41 pm" src="https://user-images.githubusercontent.com/40970507/48986618-1117a100-f0dd-11e8-80e0-9f2954eaa027.png">
